pkgs/applications/display-managers/greetd/wlgreet.nix

```diff
@@ -16,6 +18,11 @@ rustPlatform.buildRustPackage rec {
 
   cargoHash = "sha256-1ugExUtrzqyd9dTlBHcc44UrtEfYrfUryuG79IkTv2Y=";
 
+  postFixup = ''
```

Comment:
i am not against this solution, but have you tried runtimeDependencies ?

https://nixos.org/manual/nixpkgs/stable/#setup-hook-autopatchelfhook

Reply:
Oh that's great! Changed to use that instead.